common-model/src/main/java/com/sequenceiq/common/api/cloudstorage/CloudStorageBase.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.common.api.cloudstorage;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import io.swagger.annotations.ApiModel;
+
+@ApiModel
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public abstract class CloudStorageBase implements Serializable {
+
+    private S3Guard s3Guard;
```

Comment:
this field is really aws specific. as I see other fields are not really. can we somehow use a base . class here as well?

Reply:
I'll move it inside an AwsStorageParameters class and on the API it will be called aws.